runtime/descriptor.go

```diff
@@ -117,5 +119,31 @@ func makeStructFieldDescriptor(t *Type, fieldName, propertyName string) *Object
 		}
 		return ret, raised
 	}
+	if writtable {
+		setterFunc := func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+			ret := None
+			var raised *BaseException
+			if raised = checkFunctionArgs(f, fieldName, args, ObjectType, ObjectType); raised == nil {
```

Comment:
Unless it proves to be really ugly, it'd be good to use early returns in your code. If you're feeling adventurous then also change the style of getterFunc.